auth_totp/controllers/main.py

```diff
@@ -55,23 +54,38 @@ def unquote(cls, value):
 
 
 class AuthTotp(Home):
+
     @http.route()
     def web_login(self, *args, **kwargs):
-        response = super(AuthTotp, self).web_login(*args, **kwargs)
-
+        if request.params.get('login') and request.params.get('password'):
+            User = request.env["res.users"]
+            uid = User.sudo().authenticate(
+                request.params.get('db'),
+                request.params.get('login'),
+                request.params.get('password'),
+                {}
+            )
+            if uid:
+                user = User.browse(uid).sudo()
+                if user.mfa_enabled:
+                    return self.redirect(http)
```

Comment:
The redirect I have is to:

     https://<xxx>/web/None

Do you have an idea why?
From where come `http`here?

Reply:
I'm not sure about the http, but in this case shouldn't we redirect to `/auth_totp/login` ?

Reply:
ping @max3903, any idea?

Reply:
`http` is actually the http python module, the method is below. `http=http` would be cleaner but actually it doesn't need to be passed as parameter IMO.
Anyway, I still have no clue why the redirect is `/web/None`.

Reply:
Maybe you are right @leemannd that it should be redirected to `/auth_totp/login` because it happens when the secured cookie is expired.